### PR TITLE
VB-4785 - Optimise notification service

### PR DESCRIPTION
**Pull request by adaviesMOJ**
## What does this PR do?
- Move methods into their own services
- Remove unused classes / code
- Split tests into their own files (1 file per event type)
- Add handling of errors at client level to remove use of try catch at service level

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/notificationsalertsvsip/client/PrisonRegisterClient.kt [hunk 1]
```diff
@@ -17,19 +21,47 @@ class PrisonRegisterClient(
...
+        } else {
+          LOG.info("getSocialVisitContact Not Found get request $uri")
+          return@onErrorResume Mono.empty()
+        }
+      }
       .block(apiTimeout)
   }
+
+  fun isNotFoundError(e: Throwable?) =
+    e is WebClientResponseException && e.statusCode == NOT_FOUND
```
Comment (dhirajshettymoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/notificationsalertsvsip/client/PrisonRegisterClient.kt:66]: we could move this to a common Util

File: src/main/kotlin/uk/gov/justice/digital/hmpps/notificationsalertsvsip/service/SmsSenderService.kt [hunk 2]
```diff
@@ -4,118 +4,93 @@ import org.slf4j.Logger
...
-          templateVars["reference"] = visit.reference
-
-          val prisonContactNumber: String? = getPrisonSocialVisitsContactNumber(visit.prisonCode)
-          if (!prisonContactNumber.isNullOrEmpty()) {
-            templateVars["prison phone number"] = prisonContactNumber
-            templateName = SmsTemplateNames.VISIT_CANCEL
-          } else {
-            templateName = SmsTemplateNames.VISIT_CANCEL_NO_PRISON_NUMBER
-          }
+          handleCancelEvent(visit)
```
Comment (dhirajshettymoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/notificationsalertsvsip/service/SmsSenderService.kt:39]: nice! much cleaner

**Review by dhirajshettymoj — APPROVED** [review]
1 small comment, can be ignored. Nice work!

**Review by dhirajshettymoj — APPROVED** [review]
LGTM

File: src/main/kotlin/uk/gov/justice/digital/hmpps/notificationsalertsvsip/client/PrisonRegisterClient.kt (hunk 1 above)
Comment (adaviesMOJ) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/notificationsalertsvsip/client/PrisonRegisterClient.kt]: implemented

**Pull request merged**